### PR TITLE
ESQL: Merge `Configuration`

### PR DESCRIPTION
**Pull request by nik9000**
This merges esql-core's `Configuration` into esql's `EsqlConfiguration` and then renamed esql's configuration to `Configuration`. Now there's only one configuration! Not two layers, one that was copied from ql. After this we can make changes to our configuration to better suite our needs much more easilly.

It also centralizes a place to grab test configuration into `ConfigurationTestUtils` in esql's test fixtures. Most of that was scattered between a test fixture and the serialization tests.

### CODE REVIEW

File: x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/ConfigurationTestUtils.java [hunk 1]
```diff
@@ -5,58 +5,53 @@
...
+import static org.elasticsearch.test.ESTestCase.randomFrom;
+import static org.elasticsearch.test.ESTestCase.randomIntBetween;
+import static org.elasticsearch.test.ESTestCase.randomNonNegativeInt;
+import static org.elasticsearch.test.ESTestCase.randomRealisticUnicodeOfLength;
+import static org.elasticsearch.test.ESTestCase.randomZone;
 import static org.elasticsearch.xpack.esql.EsqlTestUtils.randomLiteral;
-import static org.elasticsearch.xpack.esql.session.EsqlConfiguration.QUERY_COMPRESS_THRESHOLD_CHARS;
+import static org.elasticsearch.xpack.esql.session.Configuration.QUERY_COMPRESS_THRESHOLD_CHARS;
 
-public class EsqlConfigurationSerializationTests extends AbstractWireSerializingTestCase<EsqlConfiguration> {
```
Comment (nik9000) [inline x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/ConfigurationTestUtils.java:39]: The serialization test still lives! But much of it's contents now live in this test fixture thing.

**Review by costin — APPROVED** [review]
👍

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java [hunk 2]
```diff
@@ -68,8 +78,11 @@ public EsqlConfiguration(
         assert tables != null;
     }
 
-    public EsqlConfiguration(BlockStreamInput in) throws IOException {
-        super(in.readZoneId(), Instant.ofEpochSecond(in.readVLong(), in.readVInt()), in.readOptionalString(), in.readOptionalString());
+    public Configuration(BlockStreamInput in) throws IOException {
+        this.zoneId = in.readZoneId();
+        this.now = Instant.ofEpochSecond(in.readVLong(), in.readVInt()).atZone(zoneId);
+        this.username = in.readOptionalString();
+        this.clusterName = in.readOptionalString();
```
Comment (astefan) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java:85]: I _think_ clustername is always present, we are taking it from the clusterService. So maybe not optional?

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java [hunk 3]
```diff
@@ -68,8 +78,11 @@ public EsqlConfiguration(
         assert tables != null;
     }
 
-    public EsqlConfiguration(BlockStreamInput in) throws IOException {
-        super(in.readZoneId(), Instant.ofEpochSecond(in.readVLong(), in.readVInt()), in.readOptionalString(), in.readOptionalString());
+    public Configuration(BlockStreamInput in) throws IOException {
+        this.zoneId = in.readZoneId();
+        this.now = Instant.ofEpochSecond(in.readVLong(), in.readVInt()).atZone(zoneId);
+        this.username = in.readOptionalString();
```
Comment (astefan) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java:84]: This one is always `null` with a `TODO` next to it in `TransportEsqlQueryAction`.
In SQL we used this one for a function called `USER()` to return the current user. Same goes for `clusterName` - it was used for a function called `DATABASE()`.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java [hunk 4]
```diff
@@ -57,7 +64,10 @@ public EsqlConfiguration(
         boolean profile,
         Map<String, Map<String, Column>> tables
     ) {
-        super(zi, username, clusterName);
+        this.zoneId = zi;
```
[the remark this answers is not in the document]
Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java:67]: Good catch!

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java (hunk 2 above)
Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java:85]: I can leave a TODO about it but I don't want to mix the serialization change with the refactoring.

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java (hunk 3 above)
Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java:84]: :+1:

**Review by alex-spies — APPROVED** [review]
LGTM, thanks Nik!

**Pull request merged**